spec/support/request/stripe_helper.rb

```diff
@@ -15,10 +15,9 @@ def checkout_with_stripe(guest_checkout: true, remember_card: false)
   end
 
   def fill_out_card_details
-    expect(page).to have_css("input[name='cardnumber']")
-    fill_in 'Card number', with: '4242424242424242'
-    fill_in 'MM / YY', with: "01/#{DateTime.now.year + 1}"
-    fill_in 'CVC', with: '123'
+    fill_in "stripe-cardnumber", with: '4242424242424242'
+    fill_in "exp-date", with: "01/#{DateTime.now.year + 1}"
+    fill_in "cvc", with: "123"
   end
```

Comment:
Currently I only see this commit, which changes the helper only - necessary to make the spec pass I guess.

But in a previous version of this PR there was a change on the `app/views/shared/_stripe_js.html.haml`... With this previous change it does seem to work, tested it locally:

![image](https://user-images.githubusercontent.com/49817236/156653529-dd40a929-13a9-4f4c-b72e-43149523468b.png)

Yaay! :tada:  So, could it be this change got unintended overwritten by force-pushed or so?...

Reply:
Ideally, we don't load anything from the internet, including Stripe JS code. It's great when you can run specs while being offline and the specs don't depend on third party services. Then the specs are deterministic and a lot faster.

But as you found, the mocked version is very different to the real deal. It would be great if we could better fake the real script, record real interactions and play them back in specs. I had a go at adding [Puffing Billy](https://github.com/oesmith/puffing-billy/) to another project at some point but couldn't get it working. Maybe it's easier these days... there should be a solid solution for testing Stripe.

Reply:
First, I've pushed a version that actually use the online version. 
Then, I found that it triggers some others issues, and using external sources should be avoided. So, I rollbacked, and force-pushed another solution, that make the test pass. 
I don't think we want to test stripelement with this feature test: we can by default trust them. We should only test what we are responsible for. 
I agree with @mkllnk , it could be great to improve that mocked script somehow, but it could be very time consuming.

Reply:
> I don't think we want to test stripelement with this feature test: we can by default trust them

Ok, so in that case, we skip the test automation on this form for now - thanks for clarifying.

So the change here is to use the css selector :+1: 

Merging.